File: tests/fixtures/sample.java
// Sample Java fixture for Strata smoke tests
import java.util.List;
import java.util.Optional;

public class OrderProcessor {

    private final InventoryService inventory;
    private final PaymentGateway payment;

    public OrderProcessor(InventoryService inventory, PaymentGateway payment) {
        this.inventory = inventory;
        this.payment = payment;
    }

    public OrderResult processOrder(Order order) {
        if (!validateOrder(order)) {
            return OrderResult.failure("Invalid order");
        }
        boolean reserved = inventory.reserve(order.getItems());
        if (!reserved) {
            return OrderResult.failure("Out of stock");
        }
        boolean charged = payment.charge(order.getTotal(), order.getCurrency());
        if (!charged) {
            inventory.release(order.getItems());
            return OrderResult.failure("Payment failed");
        }
        return OrderResult.success(order.getId());
    }

    private boolean validateOrder(Order order) {
        return order != null
            && order.getItems() != null
            && !order.getItems().isEmpty()
            && order.getTotal() > 0;
    }
}

interface InventoryService {
    boolean reserve(List<OrderItem> items);
    void release(List<OrderItem> items);
}

interface PaymentGateway {
    boolean charge(double amount, String currency);
    boolean refund(String transactionId);
}

class Order {
    private String id;
    private List<OrderItem> items;
    private double total;
    private String currency;

    public String getId()         { return id; }
    public List<OrderItem> getItems() { return items; }
    public double getTotal()      { return total; }
    public String getCurrency()   { return currency; }
}

class OrderItem {
    private String productId;
    private int quantity;

    public String getProductId() { return productId; }
    public int getQuantity()     { return quantity; }
}

class OrderResult {
    private final boolean success;
    private final String message;

    private OrderResult(boolean success, String message) {
        this.success = success;
        this.message = message;
    }

    public static OrderResult success(String orderId) {
        return new OrderResult(true, orderId);
    }

    public static OrderResult failure(String reason) {
        return new OrderResult(false, reason);
    }
}

File: tests/fixtures/sample.py
"""Sample Python fixture for Strata smoke tests."""


class PaymentService:
    """Handles payment processing."""

    def __init__(self, gateway):
        self.gateway = gateway

    def process_payment(self, amount: float, currency: str) -> bool:
        if not validate_amount(amount):
            return False
        charge = self.charge_card(amount, currency)
        return charge is not None

    def charge_card(self, amount: float, currency: str):
        return self.gateway.charge(amount, currency)

    def refund(self, transaction_id: str) -> bool:
        return self.gateway.refund(transaction_id)


class OrderService:
    """Manages order lifecycle."""

    def __init__(self, payment_service: PaymentService):
        self.payment = payment_service

    def place_order(self, cart, user):
        total = calculate_total(cart)
        ok = self.payment.process_payment(total, "USD")
        if ok:
            notify_user(user, "Order placed!")
        return ok


def validate_amount(amount: float) -> bool:
    return isinstance(amount, (int, float)) and amount > 0


def calculate_total(cart) -> float:
    return sum(item.price * item.quantity for item in cart.items)


def notify_user(user, message: str) -> None:
    print(f"Notify {user.email}: {message}")
